### PR TITLE
Allow CPU shares to be set when owner is system

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: src/ImageRunModal.jsx [hunk 1]
```diff
@@ -837,8 +837,19 @@ export class ImageRunModal extends React.Component {
...
-                        {(this.state.owner === systemOwner || (this.state.image && this.state.image.isSystem)) &&
-                            <FormGroup fieldId='run-image-cpu-priority' label={_("CPU shares")}>
+                        {this.state.owner === systemOwner &&
+                            <FormGroup
+                              fieldId='run-image-cpu-priority'
+                              label={_("CPU shares")}
+                              labelIcon={
+                                  <Popover aria-label={_("CPU Shares help")}
+                                      enableFlip
+                                      bodyContent={_("CPU shares determine how much CPU time a container can use relative to other running containers. Set it to a lower number for lower priority or a higher number for higher.")}>
```
Comment (martinpitt) [inline src/ImageRunModal.jsx:847]: IMHO the second sentence is rather redundant, and makes the string harder to translate. Can we trim this a bit?

    The more shares a container has, the more CPU time it gets relative to other running containers.

File: src/ImageRunModal.jsx [hunk 2]
```diff
@@ -837,8 +837,19 @@ export class ImageRunModal extends React.Component {
                             </Flex>
                         </FormGroup>
 
-                        {(this.state.owner === systemOwner || (this.state.image && this.state.image.isSystem)) &&
-                            <FormGroup fieldId='run-image-cpu-priority' label={_("CPU shares")}>
+                        {this.state.owner === systemOwner &&
```
Comment (martinpitt) [inline src/ImageRunModal.jsx:840]: Did you deliberately change the condition here?

**Review by martinpitt — CHANGES_REQUESTED** [review]

File: src/ImageRunModal.jsx (hunk 2 above)
Comment (jelly) [reply src/ImageRunModal.jsx:840]: Yes, for the new create container dialog the cpu shares option was not shown.

Comment (jelly) [reply src/ImageRunModal.jsx:840]: Ah woops, I put it in the wrong commit :)

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (garrett) [reply src/ImageRunModal.jsx:847]: How's this?

```suggestion
                                      bodyContent={_("CPU shares determine priority of running containers. Default priority is 1024. Higher numbers prioritize containers. Lower numbers decrease priority.")}>
```

**Comment by garrett** [comment]
We also talked about showing "default priority" / "higher priority" / "lower priority" next to the number. (This can be added outside of this PR.)

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (jelly) [reply src/ImageRunModal.jsx:847]: We already shown the default as 1024 as greyed out, should we mention it in the information popover a well.

Comment (garrett) [reply src/ImageRunModal.jsx:847]: It does help to explain what's going on.

Comment (jelly) [reply src/ImageRunModal.jsx:847]: Ok,`Higher numbers prioritize containers` sounds a bit vague to me, as one could read it as prioritizes all containers.

Comment (martinpitt) [reply src/ImageRunModal.jsx:847]: How about singular? "A higher number prioritizes this container"?

Comment (jelly) [reply src/ImageRunModal.jsx]: Sounds good, I made the suggested changes.

File: src/ImageRunModal.jsx [hunk 3]
```diff
@@ -844,8 +844,19 @@ export class ImageRunModal extends React.Component {
...
-                        { this.state.image && this.state.image.isSystem &&
-                            <FormGroup fieldId='run-image-cpu-priority' label={_("CPU shares")}>
+                        {this.state.owner === systemOwner &&
+                            <FormGroup
+                              fieldId='run-image-cpu-priority'
+                              label={_("CPU shares")}
+                              labelIcon={
+                                  <Popover aria-label={_("CPU Shares help")}
+                                      enableFlip
+                                      bodyContent={_("CPU shares determine priority of running containers. Default priority is 1024. A higher number prioritize this container. Lower numbers decrease priority.")}>
```
Comment (garrett) [inline src/ImageRunModal.jsx:854]: The statements should be internally consistent with each other.

For example:

```suggestion
                                      bodyContent={_("CPU shares determine the priority of running containers. Default priority is 1024. A higher number prioritizes this container. A lower number decreases priority.")}>
```

Alternate:

```suggestion
                                      bodyContent={_("CPU shares determine the priority of running containers. Default priority is 1024. Higher numbers prioritize this container. Lower numbers decrease the priority of this container.")}>
```

(FWIW: I like the first option better.)

**Review by garrett — DISMISSED** [review]
We need to fix the text a bit. It wasn't consistent with itself. I've provided some solutions.

**Comment by martinpitt** [comment]
Looks good, thanks! This will need a rebase/updated pixel test, though. Will formally approve after that.

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**